src/nvim/search.c

```diff
@@ -1101,8 +1102,8 @@ int do_search(
 
     searchstr = pat;
     dircp = NULL;
-    /* use previous pattern */
-    if (pat == NULL || *pat == NUL || *pat == dirc) {
+    // use previous pattern
+    if (pat == NULL || *pat == NUL || *pat == search_delim) {
```

Comment:
You missed some changes in this function.

Reply:
Indeed, thank you for catching that!